modules/distribution-zones/src/main/java/org/apache/ignite/internal/distributionzones/rebalance/ZoneRebalanceRaftGroupEventsListener.java

```diff
@@ -331,10 +331,14 @@ static void doStableKeySwitch(
                     )
             ).get();
 
+            // Safe to get the latest version until lwm mechanism is implemented for zone lifecycle.
+            int catalogVersion = catalogService.latestCatalogVersion();
```

Comment:
> until lwm mechanism is implemented for zone lifecycle.
I don't think so. It's still possible to remove zone without LWM.

We may discuss the way of retrieval catalog version in this case.